book/src/rfc/rfc-0-generic-parameters.md

```diff
@@ -519,6 +519,35 @@ fn main() {
 
 The current `compute_type` function can remain as it is. It stores the computed concrete types, such as `Array` or `Custom` as shown in the examples above, instead of the generic types. It creates a mapping between the expression node from the AST and the type `TypeKind`.
 
+### Function Call Instantiation
+The functions are defined as `FunctionDef`, which is an AST containing the signature and the body of the function. The body is a vector of statements, each of which is a tree of expression nodes. It is fine to have different function calls pointing to these functions' original AST, when the content of these functions doesn't change, and so are the expression nodes.
+
+However, when a function takes generic arguments, the actual arguments can result in different expression nodes. For example:
+
+```rust
+fn last(arr: [Field; N]) -> Field {
+    return arr[N - 1];
+}
+
+fn main() {
+    let arr1 = [1, 2, 3, 4, 5];
+    let arr2 = [6, 7, 8, 9];
+
+    let last1 = last(arr1); // instantiate `last` function with N = 5
+    let last2 = last(arr2); // instantiate `last` function with N = 4
+}
+```
+
+The monomorphized body of the function call for `last(arr1)` is `return arr[5 - 1]`, while the one for `last(arr2)` is `return arr[4 - 1]`. Therefore, we can't have a single expression node representing both `arr[5 - 1]` and `arr[4 - 1]` expression nodes. These functions should be instantiated with new ASTs, which are monomorphized from the original ASTs. They will be regenerated with the generic parameters being resolved with concrete values. 
```

Comment:
... and the two calls to the `last()` functions should be made to the two different monomorphized instances